Source/Core/VideoBackends/D3D12/Render.cpp

```diff
@@ -1244,6 +1242,8 @@ void Renderer::SetSamplerState(int stage, int tex_index, bool custom_tex)
 	if (tex_index)
 		stage += 4;
 
+	const auto original_hex = gx_state.sampler[stage].hex;
```

Comment:
I chose `auto` here to avoid explicitly binding the type of `hex`, so if `hex` gets changed to a `u64` then this will automatically adapt to the new type.

My original draft of this did use a local `SamplerState` instance; I opted not to go with that because it would create a merge conflict with PR #3672 but I can do that instead if you prefer.